File: Design and Programming Environments/Iteration 3/TicketManager/Services/src/main/java/Interfaces/IServer.java
package Interfaces;

import Domain.Event;
import Domain.Ticket;
import Domain.User;
import Exceptions.BasketException;
import Utils.EventChangeEvent;
import Utils.Observer;

public interface IServer {
    void login(User person,IObserver client) throws BasketException;

    void logout(User person) throws BasketException;

    Iterable<Ticket> getTickets() throws BasketException;

    Iterable<Event> getEvents() throws BasketException;

    void addEvent(Integer id,String gameName,Double gamePrice,Integer freeSeats) throws BasketException;

    void deleteEvent(Integer id) throws BasketException;

    void updateEvent(Integer id,String gameName,Double gamePrice,Integer freeSeats) throws BasketException;

    void addTicket(Integer idGame,Integer reservedSeats,String clientName) throws BasketException;

    void deleteTicket(Integer id) throws BasketException;

    Ticket getTicket(Integer id) throws BasketException;

    Event getEvent(Integer id) throws BasketException;
}

File: Second year/Design and Programming Environments/TicketManagerCSharp/Services/Service Interfaces/IServer.cs
using System.Collections.Generic;
using TicketManagerCSharp.domain;

namespace Services
{
    public interface IServer
    {
        void login(User user, IObserver client);

        void logout(User user);

        IList<Ticket> getTickets();

        IList<Event> getEvents();

        void addEvent(int id, string game_name, double game_price, int free_seats);

        void deleteEvent(int id);

        void updateEvent(int id, string game_name, double game_price, int free_seats);

        void addTicket(int id_game, int reserved_seats, string client_name);

        void deleteTicket(int id);
    }
}